packages/core/src/domain/error/trackConsoleError.ts

```diff
@@ -12,29 +12,56 @@ import { find, jsonStringify } from '../../tools/utils'
 import { callMonitored } from '../internalMonitoring'
 import { computeStackTrace } from '../tracekit'
 
-let originalConsoleError: (...params: unknown[]) => void
-
 /* eslint-disable no-console */
 export function trackConsoleError(errorObservable: Observable<RawError>) {
-  originalConsoleError = console.error
-
-  console.error = (...params: unknown[]) => {
-    const handlingStack = createHandlingStack()
-    callMonitored(() => {
-      originalConsoleError.apply(console, params)
-      errorObservable.notify({
-        ...buildErrorFromParams(params, handlingStack),
-        source: ErrorSource.CONSOLE,
-        startClocks: clocksNow(),
-        handling: ErrorHandling.HANDLED,
+  startConsoleErrorProxy().afterCall((error) => errorObservable.notify(error))
+
+  return {
+    stop: resetConsoleErrorProxy,
+  }
```

Comment:
The `stop` here is a bit confusing since it stops console tracking globally, not only where `trackConsoleError` was used. Since it is only used in tests, i think it would be better to use `resetConsoleErrorProxy` directly there.